applications/serial_lte_modem/src/slm_at_tcp_proxy.c

```diff
@@ -713,6 +740,85 @@ static void tcpcli_thread_func(void *p1, void *p2, void *p3)
 	rsp_send(rsp_buf, strlen(rsp_buf));
 }
 
+/**@brief handle AT#XTCPWHITELIST commands
+ *  AT#XTCPWHTLST=<op>[,<IP_ADDR#1>[,<IP_ADDR#2>[,...]]]
+ *  AT#XTCPWHTLST?
+ *  AT#XTCPWHTLST=?
+ */
+static int handle_at_tcp_whitelist(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+	int param_count = at_params_valid_count_get(&at_param_list);
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_short_get(&at_param_list, 1, &op);
+		if (err) {
+			return err;
+		}
+		if (op == AT_WHITELIST_SET) {
+			char address[INET_ADDRSTRLEN];
+			int size;
+
+			if (param_count > (CONFIG_SLM_WHITELIST_SIZE + 2)){
+				return -EINVAL;
+			}
+			for (int i = 0; i < CONFIG_SLM_WHITELIST_SIZE; i++) {
+				memset(ip_whitelist[i], 0x00, INET_ADDRSTRLEN);
+			}
+			for (int i = 2; i < param_count; i++) {
+				size = INET_ADDRSTRLEN;
+				err = at_params_string_get(&at_param_list, i,
+					address, &size);
+				if (err) {
+					return err;
+				} else if (!check_for_ipv4(address, size)) {
+					return -EINVAL;
+				} else {
+					memcpy(ip_whitelist[i-2], address,
+						size);
+				}
+			}
+			proxy.whitelist = true;
+			err = 0;
+		} else if (op == AT_WHITELIST_CLEAR) {
+			for (int i = 0; i < CONFIG_SLM_WHITELIST_SIZE; i++) {
```

Comment:
Or simply `memset(ip_whitelist, 0, sizeof(ip_whitelist));`

Reply:
Great, thanks for the advice